pkg/cli/subcommands/check/command.go

```diff
@@ -363,6 +365,8 @@ func run(config config.Component, cliParams *cliParams, demultiplexer demultiple
 	var checkFileOutput bytes.Buffer
 	var instancesData []interface{}
 	printer := aggregator.AgentDemultiplexerPrinter{DemultiplexerWithAggregator: demultiplexer}
+	collectorData := statuscollector.GetStatus()
+	checkRuns := collectorData["runnerStats"].(map[string]interface{})["Checks"].(map[string]interface{})
```

Comment:
I don't understand why we move this variable outside of the for. What is the purpose of this ?

Reply:
Previously for each check inside the for loop, we were calling `status.GetCheckStatusJSON(c, s)`

```go
// GetCheckStatusJSON gets the status of a single check as JSON
func GetCheckStatusJSON(c check.Check, cs *checkstats.Stats) ([]byte, error) {
	s := collector.GetStatus()

	checks := s["runnerStats"].(map[string]interface{})["Checks"].(map[string]interface{})
	checks[c.String()] = make(map[checkid.ID]interface{})
	checks[c.String()].(map[checkid.ID]interface{})[c.ID()] = cs

	statusJSON, err := json.Marshal(s)
	if err != nil {
		return nil, err
	}

	return statusJSON, nil
}

// GetCheckStatus gets the status of a single check as human-readable text
func GetCheckStatus(c check.Check, cs *checkstats.Stats) ([]byte, error) {
	statusJSON, err := GetCheckStatusJSON(c, cs)
	if err != nil {
		return nil, err
	}

	st, err := renderCheckStats(statusJSON, c.String())
	if err != nil {
		return nil, err
	}
	return []byte(st), nil
}
```

So technically, we collect the collector status information multiple times if there are multiple checks. 

These changes intend to collect the collector status information once, iterate over the check and populate the information. It mostly avoiding fetching the collector status information multiple times. Or at least that is my intention. Did I miss anything @hush-hush ?